### PR TITLE
procfs: implement parsing of NetSockstat{,6}

### LINKED ISSUE #229: Add parsing of sockstat
We currently parse `/proc/net/sockstat` in the node_exporter, it would be good to extract this functionality.

* [ ] Parse `/proc/net/sockstat`
* [ ] Parse `/proc/net/sockstat6`

Sadly, it looks like these files are not documented well in the kernel docs.

### PR DESCRIPTION
**Pull request by mdlayher**
Signed-off-by: Matt Layher <mdlayher@gmail.com>

Fixes #229.

### CODE REVIEW

File: net_sockstat.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+	"github.com/prometheus/procfs/internal/util"
+)
+
+// A NetSockstat contains the output of /proc/net/sockstat{,6} for IPv4 or IPv6,
+// respectively.
+type NetSockstat struct {
+	// Used is non-nil for IPv4 sockstat results, but nil for IPv6.
+	Used      *int
+	Protocols []NetSockstatProtocol
```
Comment (mdlayher) [inline net_sockstat.go]: The existing sockstat code in node_exporter uses a map for this, but I honestly really dislike maps in APIs. Slices and structures are almost certainly easier to work with than a map of strings.

**Comment by SuperQ** [comment]
Hrmm, there's no update to the end to end test.

**Comment by mdlayher** [comment]
We don't have those in procfs right?

**Comment by mdlayher** [comment]
Thoughts on the implementation here?

**Comment by SuperQ** [comment]
Sorry, I was confused, thought I was looking at the node_exporter repo. :facepalm:

File: net_sockstat.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+// NetSockstat6 retrieves IPv6 socket statistics.
+func (fs FS) NetSockstat6() (*NetSockstat, error) {
+	return readSockstat(fs.proc.Path("net", "sockstat6"))
+}
+
+// readSockstat opens and parses a NetSockstat from the input file.
+func readSockstat(name string) (*NetSockstat, error) {
+	f, err := os.Open(name)
```
Comment (SuperQ) [inline net_sockstat.go]: We should comment this to document that this shouldn't be converted to `util.ReadFileNoStat()` due to the possible large file size.

Comment (mdlayher) [reply net_sockstat.go]: @SuperQ actually this file in particular is quite small, so I should be using the util function. Updated now.

**Pull request merged**